apps/lib/opt.c

```diff
@@ -584,17 +580,17 @@ int opt_intmax(const char *value, ossl_intmax_t *result)
     }
     /* Ensure that the value in |m| is never too big for |*result| */
     if (sizeof(m) > sizeof(*result)
-        && (m < OSSL_INTMAX_MIN || m > OSSL_INTMAX_MAX)) {
+        && (m < INTMAX_MIN || m > INTMAX_MAX)) {
```

Comment:
Surely this first && should have been an ||...

Also... The `intmax_t m;` can't be smaller than `INTMAX_MIN` or larger than `INTMAX_MAX`, so this entire if clause is just pointless now, right? Similarly for the uintmax_t bit below.

Reply:
absolutely correct, in fact the whole `if` test goes away since the local var `m` and the parameter are the same type.  fixup for two places pushed.